### DISCUSSION TITLE
Add native Javscript SDK, alongside Python and TypeScript

### DISCUSSION THREAD

**Discussion by AwokeKnowing** [modelcontextprotocol/.github#28] [Polls]
JavaScript is the language of the browser, and TypeScript came before JS even had classes and modules, and before HTTP/2 made importing many separate files practical (vs bundling). TypeScript requires a build and is not technically native web technology. As such, there are many who prefer to build in JavaScript itself, with no build process. This is likely to expand even more with AI based applications. While TypeScript has it's place, the baseline should include JavaScript support, with other languages that 'transpile into JavaScript' as alternatives.

**Comment by jspahrsummers** [comment]
The built TypeScript SDK is all ESM, so should definitely be usable in pure JS. If you have any problems, please do file issues about it!

**Reply by AwokeKnowing** [reply]
Oh fair enough. I hadn't got a chance to figure out whether this was presented as pure esm release modules, or as a 'boilerplate' which you needed to include in a TypeScript project and build to use effectively.  I see the releases are built and examples seem to show JavaScript modules and no required TypeScript ties. I hope the examples/demos/starter code continue to be presented from a JavaScript perspective.

What I would like to see for example is https://github.com/modelcontextprotocol/create-typescript-server but with create-javascript-server, or at least just a repo to clone for a javascript server, without any cli, or build or anything required.  It should work fine to just have a js file as a server, and output these js files for new servers, without other building and tooling.

In any case, those who prefer such a thing are used to building things themselves, so as long as the main libraries are provided as ESM built releases, it should be fine.  But I think Javascript should not be an 'afterthought' because MCP is a clean self-contained protocol and has no real dependence on TypeScript, and TypeScript is more of a set of tooling for building Javascript with more "IDE" and "autocomplete" support. But as ai's write more code, it starts making more sense to have things less focused on build toolchains. Just like the Artefacts just run the code right there, ai itself should be building and modifying Javascript MCP servers, without a toolchain/build required.

In any case as long creating servers from plain released esm modules remains viable, there shouldn't be any issues.
